### PR TITLE
Adds rite, Consonance of Power to Festival Sect

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Requires https://github.com/tgstation/tgstation/pull/66418 to be merged first (it is forked from that branch as that branch reworks status effects requires for this pr). You'll see some changes from it, don't worry bout that ;] )

Consonance of Power is a song that imbues your instrument with power, making your next song applying a 10 second buff to action speeds. Listening to the song for longer slowly builds up the duration of this buff, so a full song's worth will let it last for awhile.

Nullwave Vibrato cancels out Consonance of Power, and vice versa. You can only have one buff at a time.

## Why It's Good For The Game

This was one of the coolest song ideas that ALMOST got into the initial pr but didn't due to approaching the code in a really silly weakref-heavy way. Now that the code side is much simpler and just makes sense, it can be added properly once again. At the moment antimagic is a very niche buff, and other than that the chaplain has only negative or negligible benefit effect songs, this should help the chaplain be able to contribute in a positive way that isn't just brainless healing.

## Changelog

:cl:
add: New rite for festival sect: Consonance of Power!
/:cl:

### CODE REVIEW

File: code/modules/religion/festival/instrument_rites.dm [hunk 1]
```diff
@@ -111,6 +112,35 @@
...
+/datum/religion_rites/song_tuner/power_consonance/finish_effect(atom/song_player, datum/song/song_datum)
+	for(var/mob/living/carbon/human/listener in song_datum.hearing_mobs)
+		if(!listener.mind.holy_role && listener.can_block_magic(MAGIC_RESISTANCE_HOLY, charge_cost = 1))
+			continue
+		var/obj/item/organ/stomach/ethereal/charge_target = listener.getorganslot(ORGAN_SLOT_STOMACH)
+		if(!charge_target || !istype(charge_target) || charge_target.crystal_charge >= ETHEREAL_CHARGE_FULL)
+			continue
+		to_chat(listener, span_notice("You feel recharged!"))
+		playsound(listener, 'sound/weapons/fwoosh.ogg', 75, FALSE)
+		charge_target.adjust_charge(ETHEREAL_CHARGE_FULL)
```
Comment (ATH1909) [inline code/modules/religion/festival/instrument_rites.dm:142]: remember that ethereals can be overcharged, which screws them over pretty hard

Comment (tralezab) [reply code/modules/religion/festival/instrument_rites.dm:142]: This is a setter, not adding. And it doesn't set anything if the ethereal is overcharged, so in no way does this interact with the ethereal overcharge mechanic

Comment (ATH1909) [reply code/modules/religion/festival/instrument_rites.dm:142]: > This is a setter, not adding. And it doesn't set anything if the ethereal is overcharged, so in no way does this interact with the ethereal overcharge mechanic

```
/obj/item/organ/stomach/ethereal/proc/adjust_charge(amount)
	crystal_charge = clamp(crystal_charge + amount, ETHEREAL_CHARGE_NONE, ETHEREAL_CHARGE_DANGEROUS)
```
pretty sure that ain't a setter, chief

Comment (tralezab) [reply code/modules/religion/festival/instrument_rites.dm:142]: grmgrgkmmgkgrggrgg ok ok that's gotta get fixed. i have no idea how i didn't trigger this

**Comment by tralezab** [comment]
conflict (trust nobody, not even yourself)

**Pull request closed without merging**